stake-pool/program/src/processor.rs

```diff
@@ -1410,37 +1435,136 @@ impl Processor {
             return Err(ProgramError::InsufficientFunds);
         }
 
-        create_stake_account(
-            transient_stake_account_info.clone(),
-            transient_stake_account_signer_seeds,
-            system_program_info.clone(),
-        )?;
+        let maybe_split_from_account_info =
+            if let Some(ephemeral_stake_seed) = maybe_ephemeral_stake_seed {
+                let ephemeral_stake_account_info = maybe_ephemeral_stake_account_info.unwrap();
```

Comment:
consider `zip`ing the `Option`, either in the preceding `match` or above, where these are resolved. no sense getting the auditors' hopes up with an `unwrap`!